src/Xamarin.Android.Build.Tasks/Tasks/GenerateLayoutCodeBehind.cs

```diff
@@ -194,8 +194,30 @@ void LoadWidgets (string fileName, XmlReader reader, IXmlLineInfo lineinfo,  str
 				}
 			}
 			while (reader.Read ()) {
-				LoadWidgets (fileName, reader, lineinfo, androidNS, root ?? widgetRoot, globalIdCache);
+				LoadWidgets (fileName, reader, lineinfo, androidNS, widgetRoot, globalIdCache);
+			}
+		}
+
+		static readonly Dictionary<string, Func<Widget, XmlReader, Widget>> WidgetCreators = new Dictionary<string, Func<Widget, XmlReader, Widget>> {
```

Comment:
A small nit - create the dictionary with `StringComparer.Ordinal`, it's going to be faster